### PR TITLE
Sanitise fields

### PR DESCRIPTION
**Pull request by rjkerrison**
# Sanitise everything

Fixes a bug found in #426. Relates to #551.

The Joker uses hand sanitiser

### Acceptance Criteria

- [ ] Fields which could contain non-ASCII characters and break everything are sanitised

### Testing information

Try importing with `Àine` as the product name or something.

### Pre-flight Checklist

- [ ] Testing aligns with [testing strategy][testing strategy]
- [ ] DUXD review approval
- [ ] Accessibility Reviewed
- [ ] Product owner demo

### Pre-merge Checklist

- [ ] Branch test approved

[testing strategy]: https://github.com/MHRA/products/blob/master/docs/principles/testing.md "MHRA/products testing strategy"

### CODE REVIEW

File: medicines/doc-index-updater/src/create_manager/sanitiser.rs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+//         }
+
+//         if let Some(a) = self.iter.next() {
+//             Some(SanitisedString::from(a))
+//         } else {
+//             self.iter = None;
+//             None
+//         }
+//     }
+// }
```
Comment (craiga) [inline medicines/doc-index-updater/src/create_manager/sanitiser.rs]: Why is this here?

If there's a reason for it, there should be a comment to explain.

File: medicines/doc-index-updater/src/create_manager/sanitiser.rs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        assert_eq!(
+            SanitisedString::from("newline\ntest").to_string(),
+            "newline test"
+        );
+    }
+    #[test]
+    fn sanitise_remove_non_ascii() {
+        assert_eq!(
+            SanitisedString::from("emoji🙂 ∫test").to_string(),
+            "emoji test"
```
Comment (craiga) [inline medicines/doc-index-updater/src/create_manager/sanitiser.rs]: Would be nice to include an accented character here to double-check that's working. Maybe some common punctuation outside of ASCII which could easily end up being entered into a system: a non-breaking space, en dash, left and right quotation marks, etc.

File: medicines/doc-index-updater/src/create_manager/sanitiser.rs (hunk 1 above)
Comment (rjkerrison) [reply medicines/doc-index-updater/src/create_manager/sanitiser.rs]: nope, not needed, will kill

File: medicines/doc-index-updater/src/create_manager/sanitiser.rs (hunk 2 above)
Comment (rjkerrison) [reply medicines/doc-index-updater/src/create_manager/sanitiser.rs]: great call, pushed one now

**Comment by craiga** [comment]
Accented characters in substance names aren't getting sanitised.

I ran the create worker with a local copy of the Azure SDK, which I added some logging to. Here's what it says it's receiving:

```
2020-03-27T11:19:39.005503+00:00   INFO create_service_worker{time_to_wait=10s}: azure_sdk_storage_core::rest_client: Header map: {"content-length": "14", "content-type": "application/pdf", "content-md5": "dGMIgpV14XwzMbvLAMCJiw==", "x-ms-meta-keywords": "keywords", "x-ms-meta-pl_number": "[]", "x-ms-meta-author": "author", "x-ms-meta-title": "Caf Ltt", "x-ms-meta-doc_type": "Pil", "x-ms-meta-file_name": "id124", "x-ms-meta-substance_name": "[\"CRAIGTONIUM\",\"SUBTSTANCE B\",\"CAF\",\"ZO\"]", "x-ms-meta-product_name": "[\"CRAIG DWARD ANDERSON\"]", "x-ms-meta-facets": "[\"C\",\"C, CAF\xc3\x89\",\"C, CAF\xc3\x89, CRAIG DWARD ANDERSON\",\"C, CRAIGTONIUM\",\"C, CRAIGTONIUM, CRAIG DWARD ANDERSON\",\"S\",\"S, SUBTSTANCE\nB\",\"S, SUBTSTANCE\nB, CRAIG DWARD ANDERSON\",\"Z\",\"Z, ZO\xc3\x8b\",\"Z, ZO\xc3\x8b, CRAIG DWARD ANDERSON\"]", "x-ms-blob-type": "BlockBlob", "x-ms-date": "Fri, 27 Mar 2020 11:19:39 GMT", "x-ms-version": "2018-03-28"}
```

It looks like headers like `x-ms-meta-title` are getting sanitised values (the original input was "Café Løtté", and it's "Caf Ltt" above), but `x-ms-meta-facets` gets "CAF\xc3\x89" when I send the substance "café".
